### PR TITLE
Cindyrzheng/dropclass

### PR DESCRIPTION
**Pull request by cindyrzheng**
some drop class stuff, has been weirldy merged so need to review rn

### CODE REVIEW

**Comment by 1strangequark** [comment]
This all looks good to me, Cindy.

**Comment by cindyrzheng** [comment]
> This all looks good to me, Cindy.

Rolled back to when Jonathan wrote this, and then changed an error statement.

**Comment by cindyrzheng** [comment]
Still have not tested this since Im not too sure how to get ahold of a student ID

**Comment by cindyrzheng** [comment]
Left as a draft since I'm getting a 400 bad request response. will investigate more in the morn! gotta sleep now :(

**Comment by krashanoff** [comment]
Move your URL to the end of the cURL command.

**Comment by krashanoff** [comment]
The issue you're encountering in your command is malforming the cURL command - that's why it says `curl: (3) URL using...`. That's not something from our server, but cURL. I think it's because you're using weird double-quotes, rather than standard ASCII ones. The following commands run fine for me:

image

Anyways, there were a few things wrong in the command, and I messed up a few times using it here anyways. On the left side, you can see the output of the backend, and on the right, my commands. We can see that the server complained that our request was bad a few times (400). Fixing the format by changing it to use `id` instead of `studentId`, we then we got an auth issue a few times (401) because we tried to drop the professor from the class. Changing the `id` field to the student ID, 2, the request succeeds.

File: frontend/src/api.ts [hunk 1]
```diff
@@ -320,8 +320,8 @@ export const createInvite = authorized<
  * @param onSuccess Fired on successful request.
  * @param onFailure Fired on failed request.
  */
-const dropStudent = authorized<{ classId: string; studentId: string }, void>(
-  ({ classId, studentId }) => [`/class/${classId}/drop`, { studentId }]
+export const dropStudent = authorized<{ classId: string; id: string }, void>(
+  ({ classId, id }) => [`/class/${classId}/drop`, { id }]
 );
```
Comment (krashanoff) [inline frontend/src/api.ts:325]: This isn't very idiomatic, we can keep the meaning in the function signature without breaking the code.

```suggestion
export const dropStudent = authorized<{ classId: string; studentId: string }, void>(
  ({ classId, id }) => [`/class/${classId}/drop`, { id: studentId }]
);
```

File: frontend/src/ClassView/ClassList.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+          console.log(students);
+        },
+        () => {
+          setError("can't find class");
+        }
+      );
+    }
+  }, []);
+
+  //const studentId = "1";
```
Comment (krashanoff) [inline frontend/src/ClassView/ClassList.tsx:42]:
```suggestion
```

File: frontend/src/ClassView/ClassList.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+import Container from "react-bootstrap/Container";
+import Button from "react-bootstrap/Button";
+import Table from "react-bootstrap/Table";
+import Modal from "react-bootstrap/Modal";
+import Alert from "react-bootstrap/Alert";
+
+import { dropStudent, getClass } from "../api";
+
+// Student view of the assignment
```
Comment (krashanoff) [inline frontend/src/ClassView/ClassList.tsx:14]:
```suggestion
```

**Review by krashanoff — APPROVED** [review]
Looks good following some nits!

**Pull request merged**